tfhe/src/high_level_api/integers/types/compact.rs

```diff
@@ -90,3 +92,273 @@ where
+            change_parameters(&|ct: &mut Ct| {
+                &mut ct.list.ct_list.ct_list.get_mut_lwe_ciphertext_count().0
+            }),
+            change_parameters(&|ct: &mut Ct| &mut ct.list.ct_list.ct_list.get_mut_lwe_size().0),
+        ];
+
+        for breaker_list in breaker_lists {
+            for breaker in breaker_list {
+                let mut ct_clone = ct.clone();
+
+                breaker(&mut ct_clone);
+
+                assert!(!ct_clone.is_conformant(&params));
+            }
+        }
+
+        let breakers2: Vec<&ParameterModifier<'_, Ct>> = vec![
+            &|ct: &mut Ct| ct.list.ct_list.pbs_order = PBSOrder::BootstrapKeyswitch,
+            &|ct: &mut Ct| {
+                *ct.list.ct_list.ct_list.get_mut_ciphertext_modulus() =
+                    CiphertextModulus::try_new_power_of_2(1).unwrap();
+            },
+            &|ct: &mut Ct| {
+                ct.list.ct_list.ct_list.get_mut_container().pop();
+            },
+            &|ct: &mut Ct| ct.list.ct_list.ct_list.get_mut_container().push(0),
+        ];
+
+        for breaker in breakers2 {
+            let mut ct_clone = ct.clone();
+
+            breaker(&mut ct_clone);
+
+            assert!(!ct_clone.is_conformant(&params));
+        }
+    }
+
+    #[test]
+    fn test_valid_generic_compact_integer() {
+        let config = ConfigBuilder::all_disabled()
+            .enable_default_integers()
+            .build();
+
+        let (client_key, server_key) = generate_keys(config);
+
+        let public_key = CompactPublicKey::new(&client_key);
+
+        set_server_key(server_key);
+
+        let ct = CompactFheUint8::try_encrypt(0, &public_key).unwrap();
+
+        assert!(ct.is_conformant(&PARAM_MESSAGE_2_CARRY_2_KS_PBS.into()));
+
+        let mut rng = thread_rng();
+
+        for _ in 0..10 {
+            let mut ct_clone = ct.clone();
+
+            ct_clone
+                .list
+                .ct_list
+                .ct_list
+                .get_mut_container()
+                .fill_with(|| rng.gen::<u64>());
+
+            assert!(ct_clone.is_conformant(&PARAM_MESSAGE_2_CARRY_2_KS_PBS.into()));
+
+            let mut ct_clone_expanded = ct_clone.expand();
+
+            ct_clone_expanded += &ct_clone_expanded.clone();
+        }
+    }
+
+    #[test]
+    fn test_valid_generic_compact_integer_list() {
+        let config = ConfigBuilder::all_disabled()
+            .enable_default_integers()
+            .build();
+
+        let (client_key, server_key) = generate_keys(config);
+
+        let public_key = CompactPublicKey::new(&client_key);
+
+        set_server_key(server_key);
+
+        let ct = CompactFheUint8List::try_encrypt(&[0, 1], &public_key).unwrap();
+
+        let params = (
+            PARAM_MESSAGE_2_CARRY_2_KS_PBS.into(),
+            ListSizeConstraint {
+                min_inclusive_group_count: 1,
+                max_inclusive_group_count: 1,
+                group_size: 2,
```

Comment:
shouldn't it be group_size : 1 and min/max inclusive set to 2 ?

Reply:
Sure, it's equivalent but makes more sense the way you propose
When there was no `min_inclusive_group_count` this was the only way

Reply:
Changed

Reply:
Re-opening as I'm worried for corner cases here, where a spec may validate something else but we may not have the right number of blocks ?

Could you check ?

Reply:
Not sure what you have in mind

`ListSizeConstraint::is_valid` looks correct to me

Reply:
the constraint validates for group min = 1 and group max = 1 and group size = 2

but also for

group min = 2 and group max = 2 and group size = 1

right ?

which are not the same thing at all from a data layout perspective ?

Reply:
Yes
A list of 2 cts could be used as 1 group of 2 cts or as 2 groups of 1 ct, I don´t see where is the problem

Lists of ciphertexts from a client don't have a specific data layout (only a list length), do they?

Reply:
so the other size checks probably mean we are good but I stand by what I said, let's take an LweCiphertextList from core

it could contain a single LWE ciphertext of size 2*n or two ciphertexts of size n and the layout consequences for algorithms mean that some asserts will not hold

I believe the group size may not be required for the ListSize constraint ? and we just need to get the number of entities/count which are allowed within a min/max range ?

Can we plan to add a try_new for the ListContstrainSize ? To avoid people mistakenly giving a max < min

otherwise plan the use of Rust range expressions which could solve both problems at once, wdyt ?

Reply:
the try new stuff/rust range thing can come in a follow up issue or PR once we work more on this, but I belive the ListConstraint is a bit brittle at the moment